main.py

```
@@ -139,29 +176,163 @@ async def read_root():
+                    'sector': 'N/A',
+                    'currentPrice': row['unitPrice'],  # Use unitPrice as fallback
+                    'marketCap': 'N/A',
+                    'peRatio': 'N/A',
+                    'priceChange': 'N/A',
+                    'priceChangePercent': 'N/A'
+                })
+            except Exception as e:
+                ticker_errors.append(f"{ticker}: Unexpected error - {str(e)}")
+        
+        # Store the processed data
         portfolio_storage["df"] = df
```

Ruff (0.11.9)

Undefined name `portfolio_storage`

(F821)